road/stats.py

```diff
@@ -0,0 +1,306 @@
+    output_dir = os.getcwd()
+    os.makedirs(output_dir, exist_ok=True)
+
+    metrics = ["percent_rmse", "relative_error", "est_obs_ratio"]
+    metric_fields = ["Percent RMSE", "Relative Error", "Est/Obs Ratio"]
+    file_suffixes = [
+        "percent_rmse_melted.csv",
+        "relative_error_melted.csv",
+        "est_obs_ratio_melted.csv"]
+
+    group_var_sort_orders = {
+        'AT Group': [
+            'Core/CBD',
+            'UrbBiz',
+            'Urb',
+            'Sub',
+            'All Locations'],
+        'FT Group': [
+            'Fwy/Ramp',
+            'Art',
+            'Col',
+            'Loc',
+            'All Locations'],
+        'Observed Volume': [
+            '<10k',
+            '10-20k',
+            '20-50k',
+            '>=50k',
+            'All Locations'],
+    }
+    custom_order = group_var_sort_orders.get(group_var, None)
+
+    for metric, y_field, file_suffix in zip(
+            metrics, metric_fields, file_suffixes):
+        file_path = os.path.join(output_dir, f"{file_prefix}{file_suffix}")
+
+        # Check if file exists
+        if not os.path.exists(file_path):
+            print(f"File not found: {file_path}")
+            continue
+
+        config = {
+            "$schema": "https://vega.github.io/schema/vega-lite/v5.json",
+            "data": {
+                "url": f"{file_prefix}{file_suffix}"
+            },
+            "mark": {
+                "type": "bar",
+                "stroke": "black",
+                "cursor": "pointer",
+                "tooltip": True
+            },
+            "encoding": {
+                "x": {
+                    "field": "Time Period",
+                    "type": "nominal",
+                    "sort": ["Daily", "AM", "MD", "PM", "EV", "EA"],
+                    "axis": {"title": None, "labelAngle": -90}
+                },
+                "y": {"field": y_field, "type": "quantitative"},
+                "detail": {"field": y_field},
+                "xOffset": {
+                    "field": group_var,
+                    "type": "nominal",
+                    "sort": custom_order,
+                },
+                "color": {
+                    "condition": {
+                        "test": f"datum['{group_var}'] == 'All Locations'",
+                        "value": "black"
+                    },
+                    "field": group_var,
+                    "type": "nominal",
+                    "legend": {"title": "Category"},
+                    "sort": custom_order,
+                }
+            }
+        }
+        config_file_path = os.path.join(
+            output_dir, f"{group_var.lower().replace(' ', '')}_{metric}.vega.json")
+        try:
+            with open(config_file_path, 'w') as f:
+                json.dump(config, f, indent=4)
+            print(f"Configuration for {metric} saved to: {config_file_path}")
+        except Exception as e:

```

Comment:
do we need this explicit exception? As in, would it add more info than the standard cannot write this file exception? (since I don't think we have custom exceptions for all the yaml savings either anyways)